code-conventions.md

```diff
@@ -100,8 +100,17 @@ If we need to remove or alter contributed code due to a licensing issue we will
   - When omitting brackets, still indent as if the code had brackets
   - Avoid omitting brackets if it produces hard-to-read code
 * Annotations for methods and classes are placed in lines before their declarations, one per line
+  - Annotations affecting the return type should be placed before the return type if it doesn't hurt readability.
+    ```java
+    @Nullable String motd();
+    ```
+    Is good, whereas:
+    ```java
+    @Contract("null -> null; !null -> !null") Skript instance(Plugin plugin);
+    ```
+    Is not, here the contract should be placed on the line before.
 * When there are multiple annotations, place them in order:
-  - @Override -> @Nullable -> @SuppressWarnings
+  - @Override -> @SuppressWarnings
```

Comment:
Why is this change? it was placed in the preferred position

Reply:
Because `@Nullable` will now be right before the return type, this would cause confusion. Instead we can make it:
```suggestion
  - @Override -> @Nullable (If on separate line) -> @SuppressWarnings
```

Reply:
Ideally annotations that are related to the return value should go before the value and those that are related to the method should go before the modifiers, exempli gratia:
```
@Deprecated 
@Override
@SuppressWarnings({"mywarning"}) // these have nothing to do with the value
public final @NotNull String myMethod() { // nullness is to do with the value
   ...
}
```

Reply:
Sounds good to me